aiida/manage/database/delete/nodes.py

```diff
@@ -106,21 +105,9 @@ def delete_nodes(pks, verbosity=0, dry_run=False, force=False, **kwargs):
             echo.echo('Exiting without deleting')
             return
 
-    # Recover the list of folders to delete before actually deleting the nodes. I will delete the folders only later,
-    # so that if there is a problem during the deletion of the nodes in the DB, I don't delete the folders
-    repositories = [load_node(pk)._repository for pk in pks_set_to_delete]  # pylint: disable=protected-access
-
     if verbosity > 0:
         echo.echo('Starting node deletion...')
     delete_nodes_and_connections(pks_set_to_delete)
 
-    if verbosity > 0:
-        echo.echo('Nodes deleted from database, deleting files from the repository now...')
```

Comment:
just for my understanding, if you don't delete any files from the repository here, when does it get done?

Reply:
it's done lazily. Each node now keeps a reference of the files it contains in the repository. When you delete the node, those references are deleted with it, as it is just the column `repository_metadata` of the node table. There will then be a cleaning operation on the repository that can find unreferenced objects and garbage collect them. The lazy approach is not just for performance, but it is also necessary because now files are deduplicated and so when deleting a node you cannot just delete its files, because they may be shared by other nodes as well

Reply:
But how does it know when to clean the repository, how is this triggered?

Reply:
it doesn't, this will be a manual maintenance operation launched by the user. There will be a `verdi repostiory` command with options like `backup` and `clean` etc...